mailpoet/assets/js/src/automation/listing/components/cells/more.tsx

```diff
@@ -1,29 +1,250 @@
 import { EllipsisMenu, MenuItem } from '@woocommerce/components/build';
+
+import {
+  __experimentalConfirmDialog as ConfirmDialog,
+  Button,
+} from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
-import { Workflow } from '../../workflow';
+import { useState } from 'react';
+import { ListingActions, Workflow, WorkflowStatus } from '../../workflow';
+import { deleteWorkflow, duplicate, updateWorkflowStatus } from '../../actions';
+import { Edit } from './edit';
+import { Notice } from '../../../../notices/notice';
 
 type Props = {
   workflow: Workflow;
+  listingActions: ListingActions;
 };
 
-export function More({ workflow }: Props): JSX.Element {
+type UndoButtonProps = {
+  workflow: Workflow;
+  previousStatus: WorkflowStatus;
+  onComplete?: (updatedWorkflow: Workflow) => void;
+};
+
+function UndoButton({
+  workflow,
+  previousStatus,
+  onComplete,
+}: UndoButtonProps): JSX.Element {
+  const [clicked, setClicked] = useState(false);
+  const [isBusy, setIsBusy] = useState(false);
+  return (
+    <Button
+      isLink
+      disabled={clicked}
+      isBusy={isBusy}
+      onClick={() => {
+        setClicked(true);
+        setIsBusy(true);
+        updateWorkflowStatus(
+          workflow,
+          previousStatus,
+          (updatedWorkflow) => {
+            onComplete?.(updatedWorkflow);
+          },
+          () => {
+            setIsBusy(false);
+          },
+        );
+      }}
+    >
+      Undo
+    </Button>
+  );
+}
+
+export function More({ workflow, listingActions }: Props): JSX.Element {
```

Comment:
This all feels like a huge mess right now. I'm sure it should be possible to simplify or at least extract some things.